src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncIteratorMethodToStateMachineRewriter.cs

```diff
@@ -102,6 +104,19 @@ BoundExpressionStatement generateSetResultOnPromise(bool result)
             }
         }
 
+        BoundExpressionStatement GenerateCompleteOnBuilder()
+        {
+            // Produce:
+            // this.builder.Complete();
+            return F.ExpressionStatement(
+                F.Call(
+                    F.Field(F.This(), _asyncMethodBuilderField),
+                        _asyncMethodBuilderMemberCollection.SetResult, // AsyncIteratorMethodBuilder.Complete is the corresponding method to AsyncTaskMethodBuilder.SetResult
+                        _method.IsGenericTaskReturningAsync(F.Compilation)
+                            ? ImmutableArray.Create<BoundExpression>(F.Local(_exprRetValue))
+                            : ImmutableArray<BoundExpression>.Empty));
```

Comment:
Is there an argument?

Reply:
You're correct that I re-used this call from a call to SetResult. I should have removed the argument (`SetResult` never has an argument). Thanks!

---
In reply to: [337668623](https://github.com/dotnet/roslyn/pull/39436#discussion_r337668623) [](ancestors = 337668623)